### PR TITLE
[SPARK-46111][DOCS][PYTHON] Add copyright to the PySpark official documentation.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to add the Apache Spark Foundation copyright notice to the bottom of the PySpark official documentation.

### Why are the changes needed?

Our current documentation is missing the copyright. The addition of the copyright notice is necessary to ensure compliance with the Apache Software Foundation's requirements for project documentation.

### Does this PR introduce _any_ user-facing change?

No API changes, but users will now see the Apache Spark Foundation copyright notice at the bottom of each page of the PySpark documentation as below:

## Before
Screenshot 2023-11-27 at 11 49 38 AM

## After

Screenshot 2023-11-27 at 11 35 35 AM

### How was this patch tested?

Manually build the docs and confirm.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/docs/source/_templates/spark_footer.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<p class="copyright">
+    {{copyright}} The Apache Software Foundation, Licensed under the <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License, Version 2.0</a>.
```
Comment (itholic) [inline python/docs/source/_templates/spark_footer.html:2]: FYI: I follow this copyright format from [Apache Spark official web page](https://spark.apache.org/).

**Comment by HyukjinKwon** [comment]
cc @srowen FYI

File: python/docs/source/conf.py [hunk 2]
```diff
@@ -124,7 +125,8 @@
 
 # General information about the project.
 project = 'PySpark'
-copyright = ''
+# We have our custom "spark_footer.html" template, using copyright for the current year.
+copyright = f"Copyright @ {datetime.now().year}"
```
[the remark this answers is not in the document]
Comment (itholic) [reply python/docs/source/conf.py:129]: I refer to Pandas docs but we can remove the year if we want:

Screenshot 2023-11-27 at 12 40 57 PM

Can we just remove the year?

[the remark this answers is not in the document]
Comment (srowen) [reply python/docs/source/conf.py:129]: Id leave it, it's fine

[the remark this answers is not in the document]
Comment (itholic) [reply python/docs/source/conf.py:129]: Got it. Thanks for the review!

**Review by HyukjinKwon — APPROVED** [review]

**Pull request closed without merging**